lib/jekyll/tags/highlight.rb

```diff
@@ -39,6 +40,17 @@ def initialize(tag_name, markup, tokens)
         end
       end
 
+      def parse(tokens)
+        if @lang == 'raw'
+          end_pos = tokens.rindex{|x| x =~ END_HIGHLIGHT }
```

Comment:
Why would you shift to get an index?  That said this is a detail that needs to be worried about when Liquid 4 is released because we just entered into a stable period so we probably won't be jumping on Liquid 4 bandwagon for a while even so and there is plenty more that will probably need to change.